gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/account/AccountEndpointHandler.java

```diff
@@ -0,0 +1,151 @@
+import io.gravitee.am.gateway.handler.common.utils.ConstantKeys;
+import io.gravitee.am.gateway.handler.common.vertx.web.handler.RedirectHandler;
+import io.gravitee.am.model.Domain;
+import io.gravitee.am.model.Factor;
+import io.gravitee.am.model.ReferenceType;
+import io.gravitee.am.model.User;
+import io.gravitee.am.model.common.Page;
+import io.gravitee.am.model.oidc.Client;
+import io.gravitee.am.reporter.api.audit.AuditReportableCriteria;
+import io.gravitee.am.reporter.api.audit.model.Audit;
+import io.gravitee.am.service.FactorService;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class AccountEndpointHandler {
+    private static final int DEFAULT_PAGE_SIZE = 20;
+    private static final int DEFAULT_PAGE = 0;
+    private final UserService userService;
+    private final FactorService factorService;
+    private final ActivityAuditService activityAuditService;
+    private final ClientSyncService clientSyncService;
+    private final Domain domain;
+
+    public AccountEndpointHandler(UserService userService,
+                                  FactorService factorService,
+                                  ActivityAuditService activityAuditService,
+                                  ClientSyncService clientSyncService,
+                                  Domain domain) {
+        this.userService = userService;
+        this.factorService = factorService;
+        this.activityAuditService = activityAuditService;
+        this.clientSyncService = clientSyncService;
+        this.domain = domain;
+    }
+
+    public void getUser(RoutingContext routingContext) {
+        JWT token = routingContext.get(ConstantKeys.TOKEN_CONTEXT_KEY);
+        userService.findById(token.getSub()).doOnError(err -> {
+            //todo: handle error
+        }).toSingle().subscribe(user -> {
+            routingContext.put(ConstantKeys.USER_CONTEXT_KEY, user);
+            routingContext.next();
+        });
+    }
+
+    //mounted angular
+    public void getAccount(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    //static assets
+    public void getAsset(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    public void updateDevices(RoutingContext routingContext) {
+        //TODO: waiting to impl fully
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    public void getProfile(RoutingContext routingContext) {
+        AccountResponseHandler.handleDefaultResponse(routingContext, routingContext.get(ConstantKeys.USER_CONTEXT_KEY));
```

Comment:
Using a "UserEntity" in the same way as the Management API is better in my opinion in order to filter relevant data (if there is no data to filter today, maybe it will have later)
wdyt ?

Reply:
Titouan told me to use this object until we know what we need in the future, meaning that this is an iterative process and if we need a tighter model to abstract some information we can build it when we better understand the need

Reply:
ok